src/pextlib1.0/curl.c

```diff
@@ -1435,6 +1436,61 @@ CurlPostCmd(Tcl_Interp* interp, int objc, Tcl_Obj* const objv[])
 	return theResult;
 }
 
+/**
+ * curl version subcommand entry point.
+ *
+ * @param interp		current interpreter
+ * @param objc			number of parameters
+ * @param objv			parameters
+ */
+int
+CurlVersionCmd(Tcl_Interp* interp, int objc, Tcl_Obj* const objv[])
+{
+	int theResult = TCL_OK;
+	Tcl_Obj *tcl_result = NULL;
+
+	do {
+		curl_version_info_data *theVersionInfo = curl_version_info(CURLVERSION_NOW);
+		tcl_result = Tcl_NewDictObj();
+		// info from the curl version we were built against:
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("VERSION_NOW-build", -1), Tcl_NewIntObj(CURLVERSION_NOW));
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("VERSION-build", -1), Tcl_NewStringObj(LIBCURL_VERSION, -1));
+		// runtime info from the libcurl we are actually using:
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("VERSION-runtime", -1), Tcl_NewStringObj(theVersionInfo->version, -1));
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("HOST", -1), Tcl_NewStringObj(theVersionInfo->host, -1));
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("SSL_BACKEND", -1), Tcl_NewStringObj(theVersionInfo->ssl_version, -1));
+		Tcl_Obj *tcl_prots = Tcl_NewListObj(0, NULL);
+		const char * const *prot = theVersionInfo->protocols;
+		while (*prot) {
+			Tcl_ListObjAppendElement(interp, tcl_prots, Tcl_NewStringObj(*prot, -1));
+			prot++;
+		}
+		Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("PROTOCOLS", -1), tcl_prots);
+// returning feature names requires a build-time and a runtime check:
+#if LIBCURL_VERSION_NUM >= 0x078700
+		if (theVersionInfo->age >= CURLVERSION_ELEVENTH && theVersionInfo->feature_names) {
+			Tcl_Obj *tcl_feats = Tcl_NewListObj(0, NULL);
+			const char * const *feats = theVersionInfo->feature_names;
+			while (*feats) {
+				Tcl_ListObjAppendElement(interp, tcl_feats, Tcl_NewStringObj(*feats, -1));
+				feats++;
+			}
+			Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("FEATURES", -1), tcl_feats);
+		} else
+#endif
+		{
+			// it would be possible to generate a list of feature names from the feature mask
+			// but tedious because it would involve checking for the existence of many of the 
+			// feature tokens; probably not justified for the intended use of the `curl version`
+			// command (primarily debugging).
+			Tcl_DictObjPut(interp, tcl_result, Tcl_NewStringObj("FEATURE_MASK", -1), Tcl_NewIntObj(theVersionInfo->features));
+		}
+		Tcl_SetObjResult(interp, tcl_result);
+	} while (0);
```

Comment:
Looks good to me, but why the `do { … } while (0)`? It doesn't seem like you're using `break` for control flow, and if you just want to scope the variables, just use a bare scope (although I wouldn't consider using a scope necessary here).

Reply:
Oh, *that*'s why the construct is used in other functions... I completely overlooked that aspect and just copied the non-loop as something that was apparently required ... :-/